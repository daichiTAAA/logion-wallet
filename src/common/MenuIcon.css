.MenuIcon {
    margin-right: 15px;
    border-radius: 10px;
    display: block;
    height: 32px;
    width: 32px;
    border-radius: 5px;
    margin-right: 15px;
    flex-shrink: 0;
    flex-grow: 0;
}
