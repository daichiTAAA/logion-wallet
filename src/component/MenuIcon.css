.MenuIcon {
    margin-right: 15px;
    border-radius: 10px;
}
