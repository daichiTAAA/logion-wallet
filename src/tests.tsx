import { ReactElement } from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import reactRenderer, { ReactTestRenderer, ReactTestRendererJSON, act as reactAct } from 'react-test-renderer';

export function render(element: ReactElement): null | ReactTestRendererJSON | ReactTestRendererJSON[] {
    const component = reactRenderer.create(element);
    return component.toJSON();
}

export function renderer(element: ReactElement): ReactTestRenderer {
    return reactRenderer.create(element);
}

export function shallowRender<E extends ReactElement>(element: ReactElement): E {
    const renderer = createRenderer();
    renderer.render(element);
    return renderer.getRenderOutput();
}

export function mockAccount(address: string, name: string) {
    return {
        address,
        meta: {
            name
        }
    };
}

export { reactAct as act };
