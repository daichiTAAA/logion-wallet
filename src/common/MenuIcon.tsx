import React, { CSSProperties } from 'react';

import { MenuIcon as ColorThemeMenuIcon } from './ColorTheme';
import Icon from './Icon';

import './MenuIcon.css';

export interface Props extends ColorThemeMenuIcon {
    
}

export default function MenuIcon(props: Props) {

    if(props.background === undefined && props.icon === undefined) {
        return null;
    }

    let style: CSSProperties = {
        height: props.height === undefined ? '36px' : props.height,
        width: props.width === undefined ? '36px' : props.width,
    };
    if(props.background !== undefined) {
        style['background'] = `linear-gradient(180deg, ${props.background.from} 0%, ${props.background.to} 100%)`
    }

    return (
        <div
            className="MenuIcon"
            style={style}
        >
            {
                props.icon !== undefined &&
                <Icon
                    icon={ props.icon }
                />
            }
        </div>
    );
}
