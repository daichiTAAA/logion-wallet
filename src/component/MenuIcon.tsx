import React, { CSSProperties } from 'react';

import { ColorThemeType, MenuIcon as ColorThemeMenuIcon } from './ColorTheme';

import './MenuIcon.css';

export interface Props extends ColorThemeMenuIcon {
    colorThemeType: ColorThemeType,
}

export default function MenuIcon(props: Props) {

    if(props.background === undefined && props.icon === undefined) {
        return null;
    }

    let style: CSSProperties = {
        height: props.height === undefined ? '32px' : props.height,
        width: props.width === undefined ? '32px' : props.width,
    };
    if(props.background !== undefined) {
        style['background'] = `linear-gradient(180deg, ${props.background.from} 0%, ${props.background.to} 100%)`
    }

    let iconUrl = undefined;
    if(props.icon !== undefined) {
        iconUrl = `${process.env.PUBLIC_URL}/assets/${props.icon.category}/${props.colorThemeType}/${props.icon.id}.svg`;
    }

    return (
        <div
            className="MenuIcon"
            style={style}
        >
            {
                iconUrl !== undefined &&
                <img
                    src={ iconUrl }
                    alt=''
                />
            }
        </div>
    );
}
