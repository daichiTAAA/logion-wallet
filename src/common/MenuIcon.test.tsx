import { shallowRender } from '../tests';

import MenuIcon from './MenuIcon';

test("renders with icon and background", () => {
    const result = shallowRender(
        <MenuIcon
            icon={{
                id: "iconId"
            }}
            background={{
                from: "#000000",
                to: "#ffffff",
            }}
        />
    );
    expect(result).toMatchSnapshot();
});

test("renders with icon and custom size", () => {
    const result = shallowRender(
        <MenuIcon
            icon={{
                id: "iconId"
            }}
            height='auto'
            width='auto'
        />
    );
    expect(result).toMatchSnapshot();
});
