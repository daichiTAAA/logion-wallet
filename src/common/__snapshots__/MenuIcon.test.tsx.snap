// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with icon and background 1`] = `
<div
  className="MenuIcon"
  style={
    Object {
      "background": "linear-gradient(180deg, #000000 0%, #ffffff 100%)",
      "height": "36px",
      "width": "36px",
    }
  }
>
  <Icon
    icon={
      Object {
        "id": "iconId",
      }
    }
  />
</div>
`;

exports[`renders with icon and custom size 1`] = `
<div
  className="MenuIcon"
  style={
    Object {
      "height": "auto",
      "width": "auto",
    }
  }
>
  <Icon
    icon={
      Object {
        "id": "iconId",
      }
    }
  />
</div>
`;
